### ISSUE TITLE
Select2Widget incompatible with wtforms==3.2.x

### ISSUE THREAD

**Issue by om-henners** [pallets-eco/flask-admin#2547]
After upgrading to `wtforms==3.2.1` I get the following exception when trying to use a `Select2Widget`:

```
/usr/opt/venv/lib/python3.11/site-packages/werkzeug/test.py:1162: in get
    return self.open(*args, **kw)
/usr/opt/venv/lib/python3.11/site-packages/flask/testing.py:235: in open
    response = super().open(
/usr/opt/venv/lib/python3.11/site-packages/werkzeug/test.py:1116: in open
    response_parts = self.run_wsgi_app(request.environ, buffered=buffered)
/usr/opt/venv/lib/python3.11/site-packages/werkzeug/test.py:988: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
/usr/opt/venv/lib/python3.11/site-packages/werkzeug/test.py:1264: in run_wsgi_app
    app_rv = app(environ, start_response)
/usr/opt/venv/lib/python3.11/site-packages/flask/app.py:1498: in __call__
    return self.wsgi_app(environ, start_response)
/usr/opt/venv/lib/python3.11/site-packages/flask/app.py:1476: in wsgi_app
    response = self.handle_exception(e)
/usr/opt/venv/lib/python3.11/site-packages/flask/app.py:1473: in wsgi_app
    response = self.full_dispatch_request()
/usr/opt/venv/lib/python3.11/site-packages/flask/app.py:882: in full_dispatch_request
    rv = self.handle_user_exception(e)
/usr/opt/venv/lib/python3.11/site-packages/flask/app.py:880: in full_dispatch_request
    rv = self.dispatch_request()
/usr/opt/venv/lib/python3.11/site-packages/flask/app.py:865: in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/base.py:69: in inner
    return self._run_view(f, *args, **kwargs)
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/base.py:369: in _run_view
    return fn(self, *args, **kwargs)
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/model/base.py:2175: in edit_view
    return self.render(template,
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/base.py:308: in render
    return render_template(template, **kwargs)
/usr/opt/venv/lib/python3.11/site-packages/flask/templating.py:150: in render_template
    return _render(app, template, context)
/usr/opt/venv/lib/python3.11/site-packages/flask/templating.py:131: in _render
    rv = template.render(context)
/usr/opt/venv/lib/python3.11/site-packages/jinja2/environment.py:1304: in render
    self.environment.handle_exception()
/usr/opt/venv/lib/python3.11/site-packages/jinja2/environment.py:939: in handle_exception
    raise rewrite_traceback_stack(source=source)
templates/admin/model/edit.html:3: in top-level template code
    {% from 'admin/lib.html' import extra with context %} {# backward compatible #}
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/templates/bootstrap4/admin/master.html:1: in top-level template code
    {% extends admin_base_template %}
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/templates/bootstrap4/admin/base.html:39: in top-level template code
    {% block page_body %}
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/templates/bootstrap4/admin/base.html:75: in block 'page_body'
    {% block body %}{% endblock %}
templates/admin/model/edit.html:37: in block 'body'
    {% block edit_form %}
templates/admin/model/edit.html:38: in block 'edit_form'
    {{ lib.render_form(form, return_url, extra(), form_opts) }}
/usr/opt/venv/lib/python3.11/site-packages/jinja2/runtime.py:782: in _invoke
    rv = self._func(*arguments)
templates/admin/lib.html:245: in template
    {% call form_tag(action=action) %}
/usr/opt/venv/lib/python3.11/site-packages/jinja2/runtime.py:782: in _invoke
    rv = self._func(*arguments)
templates/admin/lib.html:214: in template
    {{ caller() }}
/usr/opt/venv/lib/python3.11/site-packages/jinja2/runtime.py:782: in _invoke
    rv = self._func(*arguments)
templates/admin/lib.html:246: in template
    {{ render_form_fields(form, form_opts=form_opts) }}
/usr/opt/venv/lib/python3.11/site-packages/jinja2/runtime.py:782: in _invoke
    rv = self._func(*arguments)
templates/admin/lib.html:206: in template
    {{ render_field(form, f, kwargs) }}
/usr/opt/venv/lib/python3.11/site-packages/jinja2/runtime.py:782: in _invoke
    rv = self._func(*arguments)
templates/admin/lib.html:153: in template
    {{ field(**kwargs) | safe }}
/usr/opt/venv/lib/python3.11/site-packages/wtforms/fields/core.py:164: in __call__
    return self.meta.render_field(self, kwargs)
/usr/opt/venv/lib/python3.11/site-packages/wtforms/meta.py:64: in render_field
    return field.widget(field, **render_kw)
/usr/opt/venv/lib/python3.11/site-packages/flask_admin/form/widgets.py:28: in __call__
    return super(Select2Widget, self).__call__(field, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <flask_admin.form.widgets.Select2Widget object at 0x74b9843b6290>, field = <flask_admin.form.fields.Select2Field object at 0x74b982afaa90>
kwargs = {'class': 'form-control', 'data-allow-blank': '1', 'data-role': 'select2', 'id': 'quality'}, flags = <wtforms.fields.Flags: {optional=True}>, k = 'optional'
select_params = 'class="form-control" data-allow-blank="1" data-role="select2" id="quality" name="quality"'
html = ['<select class="form-control" data-allow-blank="1" data-role="select2" id="quality" name="quality">']

    def __call__(self, field, **kwargs):
        kwargs.setdefault("id", field.id)
        if self.multiple:
            kwargs["multiple"] = True
        flags = getattr(field, "flags", {})
        for k in dir(flags):
            if k in self.validation_attrs and k not in kwargs:
                kwargs[k] = getattr(flags, k)
        select_params = html_params(name=field.name, **kwargs)
        html = [f"<select {select_params}>"]
        if field.has_groups():
            for group, choices in field.iter_groups():
                optgroup_params = html_params(label=group)
                html.append(f"<optgroup {optgroup_params}>")
                for choice in choices:
                    val, label, selected, render_kw = choice
                    html.append(self.render_option(val, label, selected, **render_kw))
                html.append("</optgroup>")
        else:
            for choice in field.iter_choices():
>               val, label, selected, render_kw = choice
E               ValueError: not enough values to unpack (expected 4, got 3)

/usr/opt/venv/lib/python3.11/site-packages/wtforms/widgets/core.py:374: ValueError

```

Ideally this should work as it did using `wtforms==3.1.2`

Environment:

- Python version: 3.11
- Flask version: 3.0.3
- Flask-Admin version: 1.6.1

**Comment by samuelhwilliams** [comment]
Please could you provide a minimal reproduction code example to assist with debugging. Thanks :)

**Comment by samuelhwilliams** [comment]
I have just released v2.0.0a1 which should fix this issue.
